typings/internal-electron.d.ts

```diff
@@ -0,0 +1,14 @@
+/// <reference path="../electron.d.ts" />
+
+/**
+ * This file augments the Electron TS namespace with the internal APIs
+ * that are not documented but are used by Electron internally
+ */
+
+declare namespace Electron {
+  interface App {
+    setVersion(version: string): void;
+    setDesktopName(name: string): void;
+    setAppPath(path: string | null): void;
```

Comment:
```suggestion
    setAppPath(path?: string): void;
```

Reply:
This method does explicitly need `null`, see the implementation of the file path converter --> https://github.com/electron/electron/blob/master/atom/common/native_mate_converters/file_path_converter.h#L21-L34

Reply:
Using `null` to mean no argument is an anti-pattern today as new features like default parameters only work with `undefined`. It's also surprising that it *only* works with `null` and not also `undefined`. I guess I should open a new issue about changing these to only accept `undefined`? Or at minimum `undefined` too.

https://github.com/electron/electron/search?p=2&q=isnull&unscoped_q=isnull

Reply:
I agree setting null isn't exactly the JS way of doing things, but I wouldn't want it to work with `undefined` that would look super weird.  Having `setMenu()` remove the menu isn't as explicit as `setMenu(null)`.

cc @codebytere we discussed this at some point, having `removeX` instead of `setX(null)`

Reply:
Oh, sorry, I missed this was for removing the menu. I would definitely prefer an explicit method for this, or even better, use getters/setters whenever possible.

Reply:
@sindresorhus i agree, we're looking towards updating these types of things to make them more in line with modern practices